classes/requests/psr/class-truelayer-http-client.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * A PSR-18 compliant HTTP client that wraps the WordPress HTTP API.
+ *
+ * @package TrueLayer_For_WooCommerce/classes/requests/psr
+ */
+
+use KrokedilTrueLayerDeps\Nyholm\Psr7\Response;
+use KrokedilTrueLayerDeps\Psr\Http\Client\ClientInterface;
+use KrokedilTrueLayerDeps\Psr\Http\Message\RequestInterface;
+use KrokedilTrueLayerDeps\Psr\Http\Message\ResponseInterface;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class TrueLayer_Http_Client
+ */
+class TrueLayer_Http_Client implements ClientInterface {
+	/**
+	 * The log title to use for logging.
+	 *
+	 * @var string
+	 */
+	protected $log_title;
+
+	/**
+	 * The payment ID to use for the logging.
+	 *
+	 * @var string|null
+	 */
+	protected $payment_id;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param string      $log_title The log title to use for logging.
+	 * @param string|null $payment_id The payment ID to use for the logging.
+	 *
+	 * @return void
+	 */
+	public function __construct( $log_title = 'TrueLayer Request', $payment_id = null ) {
+		$this->log_title  = $log_title;
+		$this->payment_id = $payment_id;
+	}
+
+	/**
+	 * Sends a PSR-7 request and returns a PSR-7 response.
+	 *
+	 * @param RequestInterface $request The request to send.
+	 *
+	 * @return ResponseInterface
+	 */
+	public function sendRequest( RequestInterface $request ): ResponseInterface {
```

Comment:
This probably needs to throw an error here https://www.php-fig.org/psr/psr-18/#error-handling:
```A Client MUST throw an instance of Psr\Http\Client\ClientExceptionInterface if and only if it is unable to send the HTTP request at all or if the HTTP response could not be parsed into a PSR-7 response object.```